### PR TITLE
handling currency for UI display + real estate

### PR DESCRIPTION
**Pull request by nmathey**
In response to #137 
~~Include also the fix from #136~~

@lasconic let me know if you want handle currency as proposed in finary_uapi or if you prefer from a client application instead.

### CODE REVIEW

File: finary_uapi/__main__.py [hunk 1]
```diff
@@ -234,8 +234,9 @@ def main() -> int:  # pragma: nocover
             elif args["real_estates"]:
                 result = add_user_real_estates(
                     session,
-                    "rent" if args["<rent>"] else args["<category>"],
+                    "rent" if args["rent"] else args["<category>"],
```
Comment (lasconic) [inline finary_uapi/__main__.py:237]: This is in main now thanks to your PR. You can remove it here.

**Comment by nmathey** [comment]
I'm lost trying figuring out the quality check failure :( If you can help me :)

File: finary_uapi/user_real_estates.py [hunk 2]
```diff
@@ -89,7 +92,13 @@ def add_user_real_estates(
     headers = {}
     headers["Content-Length"] = str(len(data_json))
     headers["Content-Type"] = "application/json"
-    x = session.post(url, data=data_json, headers=headers)
+    if currency_code == get_display_currency_code(session):
```
Comment (lasconic) [inline finary_uapi/user_real_estates.py:95]: I'm not against this type of thing but I would prefer to have a "pure" API function with a mandatory `currency_code` and an utils function without it and doing the check in display settings and calling the pure API one. I believe it makes the maintenance easier.

**Comment by lasconic** [comment]
For quality check, run

```
poetry install `(if you didn't do it yet)`
poetry shell
./check.sh
```

File: finary_uapi/user_real_estates.py (hunk 2 above)
Comment (nmathey) [reply finary_uapi/user_real_estates.py:95]: Isn’t what I did?
I actually added currency code into add real estate function.
However as of today currency code is not mandatory in this case as using session display currency by default.
Or maybe I’m misunderstanding what you said 😅

**Comment by nmathey** [comment]
@lasconic while reading again your comment, the workaround for the real estate functions is actually already a sort of "application manipulation" .
I can propose you to remove the real estate functions modification from this PR but keep the new change display currency function since it is pure API call?

Let me know.

**Comment by lasconic** [comment]
@nmathey can you check https://github.com/lasconic/finary_uapi/pull/143 ? Does it suit your needs ?

**Pull request closed without merging**